ramanujan/enumerators/FREnumerator.py

```diff
@@ -111,26 +111,44 @@ def _improve_results_precision(self, intermediate_results, verbose=True):
         precise_intermediate_results = super()._improve_results_precision(intermediate_results, verbose)
         for i in precise_intermediate_results:
             print(i)
+        # keeping intermediate values so decedent classes could use this data
+        self.precise_intermediate_results = precise_intermediate_results
+
+        print('Running PSLQ')
         pslq_results = []
-        const = self.constants_generator[0]() # using only one constant for now.
+        # The expression PSLQ tries to find is
+        # (a + b * const) / (c + d * const) = val
+        # => a + b*const -c*val -c*d*val = 0
+        # The first two items are identical for all matches. The last two are calculated for each value
+        numer_items = [1] + [gen() for gen in self.constants_generator]
+        num_of_items = len(numer_items)
+
         for match, val, precision in precise_intermediate_results:
-            mpf_val = mpmath.mpf(val)
-            print(match)
             try:
+                mpf_val = mpmath.mpf(val)
+                denom_items = [-mpf_val * c for c in numer_items]
                 pslq_res = mpmath.pslq(
-                    [1, const, const**2, -mpf_val, -const * mpf_val, -(const**2) * mpf_val],
```

Comment:
i see that you left out the const**2 term.
please make sure that you passed all unit tests now.

if you didn't need to modify any of the tests, then it is a sign you are not testing enough :)

Reply:
I've modified the test to use a smaller list when matching results, and they all pass :)

However there is no testing for more then one constant. I'll add some test for that